3rdparty/hal_rvv/hal_rvv_1p0/addweighted.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef OPENCV_HAL_RVV_ADDWEIGHTED_HPP_INCLUDED
+#define OPENCV_HAL_RVV_ADDWEIGHTED_HPP_INCLUDED
+
+#include <riscv_vector.h>
+
+namespace cv { namespace cv_hal_rvv {
+
+#undef cv_hal_addWeighted8u
+#define cv_hal_addWeighted8u cv::cv_hal_rvv::addWeighted8u
+
+inline int addWeighted8u( const uchar* src1, size_t step1, const uchar* src2, size_t step2, 
+                        uchar* dst, size_t step, int width, int height, const void* _scalars ) {
+    const float* scalars = static_cast<const float*>(_scalars);
+    float alpha = scalars[0];
+    float beta = scalars[1];
+    float gamma = scalars[2];
+
+    for (int i = 0; i < height; ++i) {
```

Comment:
Can we ignore 2D nature and just do a single for loop from 0 to `height * width`?